locust/test/test_main.py

```diff
@@ -1425,19 +1740,47 @@ def my_task(self):
                     mocked,
                     "--headless",
                 ],
-                stdout=PIPE,
-                stderr=PIPE,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
                 text=True,
             )
-            gevent.sleep(1.9)
+
+            output = []
+
+            def condition_for_message(expected_message: str) -> Callable[[], bool]:
```

Comment:
Hmm... I think this one got too complicated. I'm sure we're saving maybe ~0.5s run time, but its not worth replacing one line of code (`gevent.sleep(1.9)`) with 20.